test/where-filter.test.js

```diff
@@ -34,6 +34,114 @@ context('whereFilter', () => {
 		result.length.should.be.equal(1);
 	});
 
+	it('SOME query returns true when array property has at least one matching entry', () => {
+		const condition = {
+			lines: {
+				some: {
+					status: 'good',
+				},
+			},
+		};
+		const data = {
+			lines: [
+				{
+					status: 'good',
+				},
+				{
+					status: 'bad',
+				},
+			],
+		};
+
+		const result = whereFilter(condition)(data);
+
+		assert.isTrue(result);
+	});
+
+	it('SOME query returns false when array property is empty', () => {
+		const condition = {
+			lines: {
+				some: {
+					status: 'good',
+				},
+			},
+		};
+		const data = {
+			lines: [],
+		};
+
+		const result = whereFilter(condition)(data);
+
+		assert.isFalse(result);
+	});
+
+	it('ALL query returns true when array property has no non-matching entries', () => {
+		const condition = {
+			lines: {
+				all: {
+					status: 'good',
+				},
+			},
+		};
+		const data = {
+			lines: [
+				{
+					status: 'good',
+				},
+				{
+					status: 'good',
+				},
+			],
+		};
+
+		const result = whereFilter(condition)(data);
+
+		assert.isTrue(result);
+	});
+
+	it('ALL query returns true when array property is empty', () => {
+		const condition = {
+			lines: {
+				all: {
+					status: 'good',
+				},
+			},
+		};
+		const data = {
+			lines: [],
+		};
+
+		const result = whereFilter(condition)(data);
+
+		assert.isTrue(result);
+	});
```

Comment:
I guess a where filter with some for an empty array will also return false.

Reply:
yes, this is based on how Array.prototype.some and Array.prototype.every (we have chosen to alias as all) work.

Reply:
Cool